src/main/java/com/microsoft/azure/functions/worker/reflect/FactoryClassLoader.java

```diff
@@ -0,0 +1,13 @@
+package com.microsoft.azure.functions.worker.reflect;
+
+import org.apache.commons.lang3.SystemUtils;
+
+public class FactoryClassLoader {
+    public ClassLoaderProvider createClassLoaderProvider(){
```

Comment:
same as above, I think this would be nice as a static `create()` method on the `ClassLoaderProvider` interface (then no need for this Factory class)